### PR TITLE
NTP Widget Region Utility Refactor

### PR DESCRIPTION
**Pull request by ryanml**
Fixes: https://github.com/brave/brave-browser/issues/11068

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
General regression testing, as there is no changed behavior

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/crypto_exchange/browser/crypto_exchange_region_util.cc [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+#include <vector>
+
+#include "brave/components/crypto_exchange/browser/crypto_exchange_region_util.h"
+
+#include "chrome/browser/profiles/profile.h"
+#include "components/country_codes/country_codes.h"
+
+namespace crypto_exchange {
+
+bool IsRegionSupported(Profile* profile, std::vector<std::string> regions,
```
Comment (ryanml) [inline components/crypto_exchange/browser/crypto_exchange_region_util.cc:16]: This should be updated to take in `PrefService*` as opposed to the whole profile

Comment (ryanml) [reply components/crypto_exchange/browser/crypto_exchange_region_util.cc]: This now accepts a `PrefService` item

File: components/crypto_exchange/browser/crypto_exchange_region_util.cc [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+#include "brave/components/crypto_exchange/browser/crypto_exchange_region_util.h"
+
+#include "components/prefs/pref_service.h"
+#include "components/country_codes/country_codes.h"
+
+namespace crypto_exchange {
+
+bool IsRegionSupported(PrefService* pref_service,
+    std::vector<std::string> regions,
```
Comment (emerick) [inline components/crypto_exchange/browser/crypto_exchange_region_util.cc:17]: This vector should probably be a const reference.

File: components/crypto_exchange/browser/buildflags/buildflags.gni [hunk 3]
```diff
@@ -1,7 +1,8 @@
 import("//brave/build/config.gni")
 import("//brave/components/binance/browser/buildflags/buildflags.gni")
+import("//brave/components/brave_together/buildflags/buildflags.gni")
 import("//brave/components/gemini/browser/buildflags/buildflags.gni")
 
 declare_args() {
-  crypto_exchanges_enabled = binance_enabled || gemini_enabled
+  crypto_exchanges_enabled = binance_enabled || gemini_enabled || brave_together_enabled
```
Comment (bbondy) [inline components/crypto_exchange/browser/buildflags/buildflags.gni:7]: I think it would make sense to rename `crypto_exchanges_enabled` given this

Comment (bbondy) [reply components/crypto_exchange/browser/buildflags/buildflags.gni:7]: maybe to `common_widget_utils_enabled`?

File: components/crypto_exchange/browser/crypto_exchange_region_util.cc (hunk 2 above)
Comment (bbondy) [reply components/crypto_exchange/browser/crypto_exchange_region_util.cc:17]: Ya: 
```
bool IsRegionSupported(PrefService* pref_service,
    const std::vector<std::string> &regions,
```

File: components/crypto_exchange/browser/crypto_exchange_region_util.h [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+#ifndef BRAVE_COMPONENTS_CRYPTO_EXCHANGE_BROWSER_CRYPTO_EXCHANGE_REGION_UTIL_H_
+#define BRAVE_COMPONENTS_CRYPTO_EXCHANGE_BROWSER_CRYPTO_EXCHANGE_REGION_UTIL_H_
+
+#include <string>
+#include <vector>
+
+class PrefService;
+
+namespace crypto_exchange {
```
Comment (bbondy) [inline components/crypto_exchange/browser/crypto_exchange_region_util.h:14]: nit: dir and namespace here seems to apply beyond crypto exchanges

File: components/crypto_exchange/browser/crypto_exchange_region_util.cc (hunk 2 above)
Comment (ryanml) [reply components/crypto_exchange/browser/crypto_exchange_region_util.cc]: done

File: components/crypto_exchange/browser/buildflags/buildflags.gni (hunk 3 above)
Comment (ryanml) [reply components/crypto_exchange/browser/buildflags/buildflags.gni]: i went for `ntp_widget_utils` for the component and renamed everything

File: components/crypto_exchange/browser/crypto_exchange_region_util.h (hunk 4 above)
Comment (ryanml) [reply components/crypto_exchange/browser/crypto_exchange_region_util.h]: see other comment

File: components/ntp_widget_utils/browser/ntp_widget_utils_region.cc [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+#include "brave/components/ntp_widget_utils/browser/ntp_widget_utils_region.h"
+
+#include "components/prefs/pref_service.h"
+#include "components/country_codes/country_codes.h"
+
+namespace ntp_widget_utils {
+
+bool IsRegionSupported(PrefService* pref_service,
+    const std::vector<std::string> regions,
```
Comment (emerick) [inline components/ntp_widget_utils/browser/ntp_widget_utils_region.cc:17]: Needs an `&` too, should be: `const std::vector<std::string>& regions`

Comment (ryanml) [reply components/ntp_widget_utils/browser/ntp_widget_utils_region.cc:17]: woops, good catch, will fix

Comment (ryanml) [reply components/ntp_widget_utils/browser/ntp_widget_utils_region.cc]: fixed

**Comment by ryanml** [comment]
Passed all platforms except for Windows, restarting for Windows

**Pull request merged**